core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/DateDirectDictionaryGenerator.java

```diff
@@ -154,14 +147,14 @@ private int generateDirectSurrogateKeyForNonTimestampType(String memberStr) {
   }
 
   private int generateKey(long timeValue) {
-    long milli = timeValue + threadLocalLocalTimeZone.get().getOffset(timeValue);
-    return (int) Math.floor((double) milli / MILLIS_PER_DAY) + cutOffDate;
+    return (int) Math.floor((double) timeValue / MILLIS_PER_DAY) + cutOffDate;
   }
 
   public void initialize() {
     if (simpleDateFormatLocal.get() == null) {
       simpleDateFormatLocal.set(new SimpleDateFormat(dateFormat));
       simpleDateFormatLocal.get().setLenient(false);
+      simpleDateFormatLocal.get().setTimeZone(TimeZone.getTimeZone("GMT"));
```

Comment:
Loading and querying both uses GMT here only for date, we should not be based on machines timezone as it gives issues